### PR TITLE
Migrate Bundler to KxS

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: plugins/package-managers/bundler/src/main/kotlin/Bundler.kt [hunk 1]
```diff
@@ -205,14 +205,14 @@ class Bundler(
         val scopes = mutableSetOf<Scope>()
         val issues = mutableListOf<Issue>()
 
-        val gemSpecs = resolveGemsMetadata(workingDir)
+        val gemsInfo = resolveGemsInfo(workingDir)
```
Comment (fviernau) [inline plugins/package-managers/bundler/src/main/kotlin/Bundler.kt:210]: Would it be better to name this either `gemInfo` or `gemInfos`? 

Putting the plural to `gems` sounds a bit odd to me, e.g. before like `gemsSpec`.

File: plugins/package-managers/bundler/src/main/kotlin/Bundler.kt [hunk 2]
```diff
@@ -404,7 +404,7 @@ class Bundler(
     }
 }
 
-internal data class GemSpec(
+internal data class GemInfo(
```
Comment (fviernau) [inline plugins/package-managers/bundler/src/main/kotlin/Bundler.kt:411]: I believe this commit should clarify which data model the class resembles. Do you have a link which could be added as KDoc here?

File: plugins/package-managers/bundler/src/main/kotlin/Model.kt [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+)
+
+@Serializable
+internal data class SpecVersion(
+    /** The version string, containing numbers and periods, such as "1.0.0.pre" for a prerelease. */
+    val version: String
+)
+
+@Serializable
+internal data class SpecDependency(
```
Comment (fviernau) [inline plugins/package-managers/bundler/src/main/kotlin/Model.kt:72]: Consider making `SpecDependency` and `SpecVersion` nested into `GemSpec`. 
This allows ommiting the prefix and spearates from the below API model classes.

File: plugins/package-managers/bundler/src/main/kotlin/Model.kt [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+internal enum class GemScope {
+    /** Dependencies required during development. */
+    @SerialName("development")
+    DEVELOPMENT,
+
+    /** Dependencies required at runtime. */
+    @SerialName("runtime")
+    RUNTIME;
+
+    override fun toString() = ":${name.lowercase()}"
```
Comment (fviernau) [inline plugins/package-managers/bundler/src/main/kotlin/Model.kt:134]: Why does the `toString()` representation start with `:` ?

File: plugins/package-managers/bundler/src/main/kotlin/Model.kt [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+@Serializable
+internal data class GemDependency(
```
Comment (fviernau) [inline plugins/package-managers/bundler/src/main/kotlin/Model.kt:138]: Also `GemScope` and `GemDependency` could be nested inside `VersionDetails` for separation and shorter name.

**Review by fviernau — CHANGES_REQUESTED** [review]

File: plugins/package-managers/bundler/src/main/kotlin/Bundler.kt (hunk 1 above)
Comment (sschuberth) [reply plugins/package-managers/bundler/src/main/kotlin/Bundler.kt:210]: I decided to do so as there is no plural of the word "information", but I somehow wanted to express that this is not just a single `GemInfo` instance.

File: plugins/package-managers/bundler/src/main/kotlin/Bundler.kt (hunk 2 above)
Comment (sschuberth) [reply plugins/package-managers/bundler/src/main/kotlin/Bundler.kt:411]: This resembles nothing official. It's a custom ORT class for passing data around. That's also why is can contain ORT-only types like `VcsInfo` and `RemoteArtifact`.

File: plugins/package-managers/bundler/src/main/kotlin/Model.kt (hunk 4 above)
Comment (sschuberth) [reply plugins/package-managers/bundler/src/main/kotlin/Model.kt:134]: Ruby uses leading colons before identifiers for [named symbols](https://stackoverflow.com/a/6337909/1127485). And these are used for the dependency type. I'll add some docs.

File: plugins/package-managers/bundler/src/main/kotlin/Bundler.kt [hunk 6]
```diff
@@ -453,17 +453,13 @@ internal data class GemSpec(
...
-            }.firstOrNull()
-
             return GemSpec(
                 node["name"].textValue(),
                 node["version"].textValue(),
                 node["homepage_uri"].textValueOrEmpty(),
                 node["authors"].textValueOrEmpty().split(',').mapToSetOfNotEmptyStrings(),
                 node["licenses"]?.toList().mapToSetOfNotEmptyStrings(),
-                description.orEmpty(),
+                node["info"].textValueOrEmpty(),
```
Comment (fviernau) [inline plugins/package-managers/bundler/src/main/kotlin/Bundler.kt]: nit: Shall we put the link ([1]) from the commit message above the `info` bit?

File: plugins/package-managers/bundler/src/main/kotlin/Model.kt [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+    val authors: List<String> = emptyList(),
+
+    /** The license(s) for the Gem. Each license must be a short name, no more than 64 characters. */
+    val licenses: List<String> = emptyList(),
+
+    /** The list of specified dependencies. */
+    val dependencies: List<Dependency> = emptyList()
+) {
+    @Serializable
+    internal data class Version(
```
Comment (fviernau) [inline plugins/package-managers/bundler/src/main/kotlin/Model.kt:65]: Could it be that `internal` is inherited from outside class and therefore redundant? (I'm not sure)

Comment (sschuberth) [reply plugins/package-managers/bundler/src/main/kotlin/Model.kt]: I've checked and indeed the `internal` from the nested classes can be dropped without making them publicly accessible.

**Review by fviernau — APPROVED** [review]

**Pull request merged**